core/src/main/java/org/ehcache/RuntimeConfiguration.java

```diff
@@ -0,0 +1,184 @@
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * @author rism
+ */
+public class RuntimeConfiguration<K, V> implements CacheRuntimeConfiguration<K, V>, InternalRuntimeConfiguration<K, V> {
+
+  private final Collection<ServiceConfiguration<?>> serviceConfigurations;
+  private final CacheConfiguration<? super K, ? super V> config;
+  private final Class<K> keyType;
+  private final Class<V> valueType;
+  private final EvictionVeto<? super K, ? super V> evictionVeto;
+  private final EvictionPrioritizer<? super K, ? super V> evictionPrioritizer;
+  private final ClassLoader classLoader;
+  private final Expiry<? super K, ? super V> expiry;
+  private final ResourcePools resourcePools;
+  private final CacheEventNotificationService<K, V> eventNotificationService;
+  private Store<K, V> store;
+  final StoreListener<K, V> storeListener;
+  private final List<CacheConfigurationListener> cacheConfigurationListenerList
+      = new CopyOnWriteArrayList<CacheConfigurationListener>();
+
+  public RuntimeConfiguration(CacheConfiguration<K, V> config, Store<K, V> store,
+                              CacheEventNotificationService<K, V> eventNotifier) {
+    this.config = config;
+    this.serviceConfigurations = copy(config.getServiceConfigurations());
+    this.keyType = config.getKeyType();
+    this.valueType = config.getValueType();
+    this.store = store;
+    this.evictionVeto = config.getEvictionVeto();
+    this.evictionPrioritizer = config.getEvictionPrioritizer();
+    this.classLoader = config.getClassLoader();
+    this.expiry = config.getExpiry();
+    this.resourcePools = config.getResourcePools();
+    this.eventNotificationService = eventNotifier;
+    this.storeListener = new StoreListener<K, V>();
+    this.storeListener.setEventNotificationService(eventNotifier);
+  }
+
+  @Override
+  public synchronized void updateResourcePools(ResourcePools pools){
+    TreeSet<ResourceType> resourceTypes = new TreeSet<ResourceType>();
+    Collections.addAll(resourceTypes, ResourceType.Core.values());
+    Iterator<ResourceType> resourceTypeIterator = resourceTypes.descendingIterator();
+    while (resourceTypeIterator.hasNext()) {
+      ResourceType currentResourceType = resourceTypeIterator.next();
+      if (config.getResourcePools().getPoolForResource(currentResourceType) != null
+          && pools.getPoolForResource(currentResourceType) != null
+          && config.getResourcePools().getPoolForResource(currentResourceType).getType()
+          == pools.getPoolForResource(currentResourceType).getType()) {
+        Long oldValue = config.getResourcePools()
+            .getPoolForResource(currentResourceType)
+            .getSize();
+        Long newValue = pools.getPoolForResource(currentResourceType).getSize();
+        fireCacheConfigurationChange(CacheConfigurationProperties.UPDATESIZE, oldValue, newValue);
```

Comment:
While I understand your question, I have no clue how this code is supposed to work even when there are no exceptions.
What will be the difference for an event updating the heap resource and one the disk resource?